modules/dataquery/static/schema.yml

```diff
@@ -0,0 +1,353 @@
+      responses:
+        '200':
+          description: Successfully operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/AllQueries'
+        '403':
+          description: Permission denied
+    post:
+      description: Create a new query and return the QueryID. If the same query (fields and criteria) already exists, the same QueryID will be returned instead of a new one being created.
+      requestBody:
+        description: A Query object (without the QueryID)
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Query'
+      responses:
+        '200':
+          description: Query successfully created
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  QueryID:
+                    type: integer
+        '400':
+          description: An invalid body was supplied
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  error:
+                    type: string
+  /queries/{QueryID}:
+    parameters:
+      - name: QueryID
+        in: path
+        description: the QueryID returned by posting to /queries
+        required: true
+        style: simple
+        schema:
+          type: integer
+    get:
+      responses:
+        '200':
+          description: The Query was successfully retrieved
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Query'
+        '403':
+          description: Permission denied
+    patch:
+      parameters:
+        - name: share
+          in: query
+          style: spaceDelimited
+          schema:
+            type: boolean
+          
+        - name: unshare
+          in: query
+          style: spaceDelimited
+          schema:
+            type: boolean
+        - name: star
+          in: query
+          style: spaceDelimited
+          schema:
+            type: boolean
+        - name: unstar
+          in: query
+          style: spaceDelimited
+          schema:
+            type: boolean
+        - name: type
+          in: query
+          style: spaceDelimited
+          schema:
+            type: string
+            enum: ['top', 'untop', 'dashboard']
+        - name: name
+          in: query
+          style: pipeDelimited
+          schema:
+            type: string
+      responses:
+        '400':
+          description: Bad request. Most likely caused by one or more incompatible patches to be done at the same time.
+  /queries/{QueryID}/run:
+    parameters:
+      - name: QueryID
+        in: path
+        description: the QueryID returned by posting to /queries
+        required: true
+        style: simple
+        schema:
+          type: integer  
+    get:
```

Comment:
I think post would make more sense. This is creating a queryRun so I don't think it is safe (as in rfc2616 safe)

Reply:
returning 201 Created with location header set to  `/queries/{QueryID}/run/{QueryRunID}` would be good as well.

Reply:
how do you denote "Location header" in swagger?

Reply:
actually that can't be done right now because that end point is not implemented so it would mean it's impossible to get results. Once it is implemented, the schema could be updated.

Reply:
'201':
          description: The visit was created successfully
          headers:
            Content-Location:
              description: The URL of the new visit.
              style: simple
              explode: false
              schema:
                type: string